Client/game_sa/CBuildingsPoolSA.cpp

```diff
@@ -98,6 +95,17 @@ CBuilding* CBuildingsPoolSA::AddBuilding(CClientBuilding* pClientBuilding, uint1
 
     // Add building in world
     auto pBuildingSA = new CBuildingSA(pBuilding);
+
+    // Edge case for the traincross2 (1374) model
+    if (pBuilding->HasMatrix())
+        pBuildingSA->ReallocateMatrix();
```

Comment:
I didn't follow the conversations about this problem, but why not just deallocate any game-allocated matrix by default in every case? Does the game expect this model to have a matrix the whole time? If not, then we don't need `ReallocateMatrix` at all.

Reply:
> Does the game expect this model to have a matrix the whole time?

No. The game allocates a static matrix for buildings that have `x` or `y` rotation in `CVector4D` and always for the traincross2 model. 
The engine uses the same storage for dynamically allocated matrixes and static allocated matrixes. It leads to two edge cases:
1) The game cannot allocate a static matrix for a building. This causes crashes.
2) The game cannot allocate a new dynamic matrix for an object when the storage has no place for it. It may cause rotation bugs for some objects.
3) Performace drop when the game recalcutes dynamic matrixes

I want to avoid this by using custom unlimited matrix storage for scripted buildings.

Reply:
In conclusion we should always deallocate the game's matrix here?

Reply:
Yes